runtime/message.cpp

```diff
@@ -501,7 +513,11 @@ size_t Message::encode(char **hmsg_store) const
 size_t Message::encode(f8String& to) const
 {
 	char output[FIX8_MAX_MSG_LENGTH + HEADER_CALC_OFFSET], *ptr(output);
-	const size_t msgLen(encode(&ptr));
+	// We don't know what amount of space encode of single field might take, so we assume that
+	//  * tag and separators will take up to MAX_MSGTYPE_FIELD_LEN
+	//  * value will take up to FIX8_MAX_FLD_LENGTH
```

Comment:
I would suggest extending the comment with the actual meaning of `endOfSafeEncode`, that is that after this pointer encoding value may overflow the buffer.